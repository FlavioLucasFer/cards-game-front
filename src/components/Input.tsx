import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    label: PropTypes.string,
    labelClassname: PropTypes.string,
};

export type InputProps = React.HTMLProps<HTMLInputElement> & PropTypes.InferProps<typeof propTypes>;

const Input: React.FC<InputProps> = props => {
    const {
        label,
        labelClassname,
        id = '',
        ...inputProps
    } = props;

    return (
        <div>
            {label &&
                <label
                    htmlFor={id}
                    className={`form-label ${labelClassname || ''}`}
                >
                    {label}
                </label>
            }
            <div className='input-group mb-3'>
                <input
                    id={id}
                    className={`form-control ${inputProps?.className || ''}`}
                    {...inputProps}
                />
            </div>
        </div>
    );
}

Input.propTypes = propTypes;

export default Input;
