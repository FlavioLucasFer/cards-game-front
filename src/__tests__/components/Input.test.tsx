import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Input from '../../components/Input';

configure({ adapter: new Adapter() });

describe('Input tests', () => {
    it('should render properly', () => {
        const wrapper = shallow(<Input />);

        expect(wrapper.debug()).toMatchSnapshot();
    });
});
