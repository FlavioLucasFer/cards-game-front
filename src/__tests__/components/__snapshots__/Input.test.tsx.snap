// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input tests should render properly 1`] = `
"<div>
  <div className=\\"input-group mb-3\\">
    <input id=\\"\\" className=\\"form-control \\" />
  </div>
</div>"
`;
